automl/automl-script/script.whizzml

```diff
@@ -0,0 +1,126 @@
+(log-info "Checking excluded fields")
+(check-excluded-fields train-dataset automl-execution excluded-fields)
+(log-info "Checking input dataset fields")
+(check-datasets-fields train-dataset validation-dataset test-dataset)
+
+;; Whether we should resources from a given execution
+(define reuse-resources (not (= (resource-type train-dataset) "dataset")))
+(log-featured "Feature generation")
+(log-info "Obtaining unsupervised models")
+;; Returns a list of unsupervised-models from a dataset
+(define (create-unsupervised-models dataset excluded)
+  (let (params {"excluded_fields" (field-ids-from-names excluded dataset)}
+        objective-id (dataset-get-objective-id dataset)
+        lev-params (merge params {"rhs_predicate" [{"field" objective-id}]
+                                  "search_strategy" "leverage"})
+        lift-params (merge params {"rhs_predicate" [{"field" objective-id}]
+                                   "search_strategy" "lift"})
+        all-models [(create-unsupervised "association" dataset lev-params false)
+                    (create-unsupervised "association" dataset lift-params false)
+                    (create-unsupervised "cluster" dataset params true)
+                    (create-unsupervised "anomaly" dataset params true)
+                    (create-unsupervised "pca" dataset params true)
+                    (create-unsupervised "topicmodel" dataset params true)])
+    (remove-false all-models)))
+
+;; Returns a list of unsupervised model from an execution output
+(define (get-unsupervised-models automl-exec)
+  (output-from-exec automl-exec "unsupervised-models"))
+;; List will all the created unsupervised models
+(define unsupervised-models
+  (if reuse-resources
+    (get-unsupervised-models automl-execution)
+    (create-unsupervised-models train-dataset excluded-fields)))
+
+(log-info "Generating new features from unsupervised models")
+(define (feature-generation dataset-id model-list excluded)
+  (let (excluded (remove-false (field-ids-from-names excluded dataset-id))
+        objective-id (dataset-get-objective-id dataset-id)
+        name (resource-name dataset-id)
+        feat-gen (lambda (m b)
+                   (unsupervised-feature-gen dataset-id model-list name m b))
+        cluster-fields (feat-gen "cluster" "batchcentroid")
+        anomaly-fields (feat-gen "anomaly" "batchanomalyscore")
+        topic-fields (feat-gen "topicmodel" "batchtopicdistribution")
+        pca-fields (feat-gen "pca" "batchprojection")
+        dataset-assoc (assoc-feature-gen dataset-id  model-list name excluded)
+        all-fields [cluster-fields anomaly-fields topic-fields pca-fields]
+        all-ds (cons dataset-assoc (map batch-output-ds all-fields)))
+    (feature-generation-dataset all-ds name objective-id)))
+
+(define extended-datasets
+  (map (lambda(d) (when (= (resource-type d) "dataset")
+                    (log-info " - Extending dataset: " (resource-name d))
```

Comment:
I guess is not very important, but jao asked me to indent things after when and if aligned with the condition parenthesis